homeassistant/components/sensor/__init__.py

```diff
@@ -729,7 +704,16 @@ def _custom_unit_or_undef(
     @callback
     def async_registry_entry_updated(self) -> None:
         """Run when the entity registry entry has been updated."""
-        self._sensor_option_precision = self._custom_precision_or_none()
+        self._async_read_entity_options()
+
+    @callback
+    def _async_read_entity_options(self) -> None:
+        """Read entity options from entity registry.
+
+        Called when the entity registry entry has been updated and before the sensor is
+        added to the state machine.
+        """
+        self._sensor_option_display_precision = self._suggested_precision_or_none()
```

Comment:
Do we still need this? It's only used in the frontend.

Reply:
It's used to no longer round in core when we're doing unit conversion if the user has set the display precision

https://github.com/home-assistant/core/blob/7991c4e2a09a832947b05fa47477cf090ad92eae/homeassistant/components/sensor/__init__.py#L591-L616

Reply:
I added a comment to make it a bit clearer.